apis/dscinitialization/v1/servicemesh_types.go

```diff
@@ -0,0 +1,28 @@
+package v1
+
+import operatorv1 "github.com/openshift/api/operator/v1"
+
+// ServiceMeshSpec configures Service Mesh.
+type ServiceMeshSpec struct {
+	// +kubebuilder:validation:Enum=Managed;Removed
+	// +kubebuilder:default=Removed
+	ManagementState operatorv1.ManagementState `json:"managementState,omitempty"`
+	// Mesh holds configuration of Service Mesh used by Opendatahub.
+	Mesh MeshSpec `json:"mesh,omitempty"`
```

Comment:
I indeed do not like too much the structure here. For Serverless it makes more sense, given `serving` is like a sub-component.

However, for SM my thought is that it is better a plain structure rather than trying to add `spec` or `mesh` or any sub-grouping of fields. 

I can change to whatever you like more.

Reply:
I am leaning towards `spec`. `serviceMesh.mesh` is odd and I'm sorry for the poor choice of names here.

Reply:
Changed, having settled on `ControlPlane` naming.